### ISSUE TITLE
Error message and crashes after applying the Sidecar patch

### ISSUE THREAD

**Issue by blueskies7777** [pookjw/SidecarPatcher#4]
Hi,
I just tried to apply the latest SidecarPatcher version on my late 2013 Macbook pro. 
Something went wrong, I received an error message. Still, after patching, I was able see the Sidecar symbol in settings, and it also showed my iPad Pro as option, but it did not work, no reaction on the iPad. Instead many other programs crashed, because - according to  error log messages - SidecarCore was causing Notes and Airmail to crash. 
So I did a rollback.
Now I have two questions:
1.  Is the sidecar patch visible only while csrutil is disabled? I am asking, because after enabling csrutils again, Sidecar was removed from settings. I would not  like to disable security for good.
2. What does the following error message mean? Did I do something wrong? I attach all commands I've entered:

Marios-Macbook:~ n$ chmod +x /Applications/SidecarPatcher 
Marios-Macbook:~ n$ sudo /Applications/SidecarPatcher 
Password:
/System/Library/PrivateFrameworks/SidecarCore.framework/Versions/A//SidecarCore: replacing existing signature
xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
/System/Library/PrivateFrameworks/SidecarCore.framework/Versions/A//SidecarCore: the codesign_allocate helper tool cannot be found or used
Reboot your macOS.

Thanks in advance for your reply!

Best regards
Mario

**Comment by pookjw** [comment]
Question 2:

1. Install Command Line Tools: `xcode-select --install`

2. Replace `/System/Library/PrivateFrameworks/SidecarCore.framework/Versions/A/SidecarCore` to original you backed up.

3. Run SidecarPatcher again.

If you didn't make a backup and already patched... then

1. Install Command Line Tools: `xcode-select --install`

2. Sign patched SidecarCore: `sudo codesign -f -s - /System/Library/PrivateFrameworks/SidecarCore.framework/Versions/A/SidecarCore`

3. Set permission as 755: `sudo chmod 755 /System/Library/PrivateFrameworks/SidecarCore.framework/Versions/A/SidecarCore`

Question 1 and Crashing many apps: I know that issue, but I don't have Sidecar-unsupported Mac as I said. So I don't know exactly why it happens, and how to fix it. My assumption is: If SidecarCore isn't signed, occurs issue you said. Could you try *Question 2*'s answer and give the result?

**Comment by blueskies7777** [comment]
Hello back,
Thank you very much for your help - very much appreciated!
I followed your instructions and it worked this time, I am able to connect the iPad with Sidecar, without app crashes.
Turns out, that in Sidecar modus the iPad screen is flickering, especially when changing programs on the Mac, and the view is not sharp in general. Not really usable.
So I think now that there was a real reason, why Apple is not supporting a Late 2103 MacBook Pro  - and not only because they just want to make more money by pushing people to buy new machines;))

Or do have an idea how to tackle the flickering?

Kind regards from Switzerland
Mario


Am 27. September 2019 um 19:01:50, Jinwoo Kim (notifications@github.com) schrieb:

Question 2:

Install Command Line Tools: xcode-select --install

Replace /System/Library/PrivateFrameworks/SidecarCore.framework/Versions/A/SidecarCore to original you backed up.

Run SidecarPatcher again.

If you didn't make a backup and already patched... then

Install Command Line Tools: xcode-select --install

Sign patched SidecarCore: sudo codesign -f -s - /System/Library/PrivateFrameworks/SidecarCore.framework/Versions/A/SidecarCore

Set permission as 755: sudo chmod 755 /System/Library/PrivateFrameworks/SidecarCore.framework/Versions/A/SidecarCore

Question 1 and Crashing many apps: I know that issue, but I don't have Sidecar-unsupported Mac as I said. So I don't know exactly why it happens, and how to fix it. My assumption is: If SidecarCore isn't signed, occurs issue you said. Could you try Question 2's answer and give the result?

**Comment by pookjw** [comment]
I have no idea about it. Anyway, thanks for giving the result.

**Comment by coodoo** [comment]
Same flickering result here on Mid-2014 MacBook Pro with iPad Pro 1st gen, thanks for the patch anyway! 👍

**Comment by pookjw** [comment]
@coodoo Does it work on 10.15 official release?

**Comment by lucien-guoheng** [comment]
> @coodoo Does it work on 10.15 official release?

for me, it works but only in wired mode.
My devices are MBP 2016 and iPad Air 2
but there is a little bit difference from beta version, that is in the official version, if you don't connect them via USB, you won't see any device in the status bar (but in beta version, it shows device but when you try to connect it shows error code 32002).
In wired mode, it works perfectly, I truly hope that the new version of this tweak can fix the issue of sidecar in wireless mode.

**Comment by coodoo** [comment]
@pookjw Yes, it worked with the official release, but the screen was flickering so bad that it's barely usable, guess there's indeed a good reason for Apple to disable support for older macs 😅

**Comment by bsormagec** [comment]
> > @coodoo Does it work on 10.15 official release?
> 
> for me, it works but only in wired mode.
> My devices are MBP 2016 and iPad Air 2
> but there is a little bit difference from beta version, that is in the official version, if you don't connect them via USB, you won't see any device in the status bar (but in beta version, it shows device but when you try to connect it shows error code 32002).
> In wired mode, it works perfectly, I truly hope that the new version of this tweak can fix the issue of sidecar in wireless mode.

how can we connect with wired mode nothing happening when i connect with wired :(
